### DISCUSSION TITLE
[Request] Option or program to restore Windows 7's intensity slider to automatically adjust the ColorBalance layers according the the Intensity slider's alpha value

### DISCUSSION THREAD

**Discussion by TheInsane101** [Maplespe/DWMBlurGlass#334] [Ideas]
UPDATE: [@CatmanFan has just written a standalone program](https://github.com/CatmanFan/DWMBG_AeroCalculator) that does the calculations and even writes them to DWMBG's config. However, my request for this to be added to DWMBG natively still stands.

@WackyIdeas made mathematical functions for how Windows 7 uses the alpha value from the Intensity slider in the colours CPL to determine the ColorizationColorbalance (formerly called Primary Colorbalance), ColorizationAfterglowbalance (formerly Secondary Colorbalance) and ColorizationBlurbalance values, and put it in a [graph](https://www.desmos.com/calculator/t5wh1idedh).

That graph was made before the more accurate formula for 7's Aero was recreated and added to DWMBG 2.2. Now that we have it, it would be nice and convenient if someone created a small program that simply has its own intensity slider that you can adjust, the value of which the program then uses as the x value in each of the 3 ColorizationBalance's functions and then returns the corresponding values for the user to put into DWMBG's GUI or config. 

I put together a concept image to better explain what I am thinking of. The top left is if the option were integrated into the DWMBG GUI, and the bottom right is if it came in its own program, making it an "Aero calculator," which I think would be quicker, simpler and easier [UPDATE: the latter has already been done by CatmanFan, but I'm keeping it here for archival purposes]:
![Aero parameters v2](https://github.com/Maplespe/DWMBlurGlass/assets/104862146/f6651f32-917f-4393-9fbf-5b41a2561699)
(On a side note, it would be helpful to have an additional window preview just for the Aero option, as shown above, since the way it is set now makes the current preview box only useful for all other non-Aero blur effects.)

Maybe it could even be incorporated into DWMBG itself later on, or, even more ambitiously, be programmed to grab the opacity balue directly from the registry when you use Windows 10's original intensity slider, indirectly making it functional again! This is much more ambitious, and personally I would be satisfied with just the simple program that does the calculations for you.

This would be especially helpful to possibly enable the Aero effects to change when setting the system's accent colour to change to match the desktop background (the auto-colour option from Windows 8), so you'd get back the Aero functionality that existed in preview builds of Windows 8 :)

This would only be useful for people like me who want to stick to 7's Aero's rules as if it's exactly 7 again, so I understand if others feel it's not worth it to go through the trouble of programming this. I would try programming this myself, but not only do I have absolutely _zero_ experience with coding, never having learned anything, but I'm currently very busy so I wouldn't even have the time to do this even if I knew how to, so I understand if nobody else feels the need to do this. It would be convenient, cool and fun though!

**Comment by kfh83** [comment]
> @WackyIdeas made formulae for how Windows 7 uses the alpha value from the Intensity slider in the colours CPL to determine the ColorizationColorbalance (formerly called Primary Colorbalance), ColorizationAfterglowbalance (formerly Secondary Colorbalance) and ColorizationBlurbalance values, and put it in a [graph](https://www.desmos.com/calculator/t5wh1idedh).
> 
> That graph was made before the more accurate formula for 7's Aero was recreated and added to DWMBG 2.2. Now that we have it, it would be nice and convenient if someone created a small program that simply has its own intensity slider that you can adjust, the value of which the program then uses as the x value in each of the 3 ColorizationBalance's functions and then returns the corresponding values for the user to put into DWMBG's GUI or config.
> 
> I put together a concept image to better explain what I am thinking of. The top left is if the option were integrated into the DWMBG GUI, and the bottom right is if it came in its own program, making it an "Aero calculator," which I think would be quicker, simpler and easier: ![Aero parameters v2](https://private-user-images.githubusercontent.com/104862146/332474285-f6651f32-917f-4393-9fbf-5b41a2561699.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzMDg1NDYsIm5iZiI6MTcxNjMwODI0NiwicGF0aCI6Ii8xMDQ4NjIxNDYvMzMyNDc0Mjg1LWY2NjUxZjMyLTkxN2YtNDM5My05ZmJmLTViNDFhMjU2MTY5OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyMVQxNjE3MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NTQ5NDgxNGNjNTRmOWU1NTExYTJhY2ZhYWY4MDg4MzYxM2U2MjNlOWI5YWMzZTc1YzRlMDYxZjQ1MTNkMGIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fx9kvbvpL5CY1b545ZO66dLBaXpMpvpii2JLicBdn5w) (On a side note, it would be helpful to have an additional window preview just for the Aero option, as shown above, since the way it is set now makes the current preview box only useful for all other non-Aero blur effects.)
> 
> Maybe it could even be incorporated into DWMBG itself later on, or, even more ambitiously, be programmed to grab the opacity balue directly from the registry when you use Windows 10's original intensity slider, indirectly making it functional again! This is much more ambitious, and personally I would be satisfied with just the simple program that does the calculations for you.
> 
> This would be especially helpful to possibly enable the Aero effects to change when setting the system's accent colour to change to match the desktop background (the auto-colour option from Windows 8), so you'd get back the Aero functionality that existed in preview builds of Windows 8 :)
> 
> This would only be useful for people like me who want to stick to 7's Aero's rules as if it's exactly 7 again, so I understand if others feel it's not worth it to go through the trouble of programming this. I would try programming this myself, but not only do I have absolutely _zero_ experience with coding, never having learned anything, but I'm currently very busy so I wouldn't even have the time to do this even if I knew how to, so I understand if nobody else feels the need to do this. It would be convenient, cool and fun though!

The intensity slider on the CPL page has been broken since atleast some early version of Windows 10, it doesn't set any value. You can notice this by going into the registry and looking at ColorizationColor as you change the slider and save, it'll always be 6b, and once you re-enter the color settings page it resets its position. 

However, it would indeed be convenient to have it in the DWMBlurGlass GUI, because i doubt anyone knows what any of the sliders mean or do.

**Reply by TheInsane101** [reply]
Oh ok, that's good to hear. Thank you. I think even just making a small, separate program whose only purpose is to calculate and give you the three Balance values for the intensity you choose, and then you add the calculated values to DWMBG manually, is convenient enough, at least for me. How does that sound?

**Reply by kfh83** [reply]
Matter of fact @wackyideas already wrote a header that returns 3 values when you input a value. Not sure if i can post it here, but yeah.

**Reply by TheInsane101** [reply]
Oh, are you talking about [this?](https://www.desmos.com/calculator/eer8gj14b3) Or [this?](https://docs.google.com/spreadsheets/d/e/2PACX-1vSlfNz2ZSygjlJhDehM7L_pkEQgMPf5BCpY1BcSPO2JFvg02EC4LeIURORNSGlo_Btt-QrRIPK69V-M/pubhtml#) I only know about these functions because of his comments under your "7 Colorization research" thread.
Or is it something private?

**Reply by WackyIdeas** [reply]
The code is open source and can be found [here](https://gitgud.io/wackyideas/kde-effects-forceblur/-/blob/master/src/wackyfunc.h?ref_type=heads), I used it for implementing the entire Aero glass effect on KDE Plasma, so feel free to use it.

**Reply by kfh83** [reply]
^ I could possibly write up a config entry from 0-217 that uses the header but i dont have my source tree setup nor visual studio.

**Reply by TheInsane101** [reply]
No need anymore, CatmanFan JUST made it already only 17 minutes ago lmao! https://github.com/CatmanFan/DWMBG_AeroCalculator

**Reply by kfh83** [reply]
Nice work. Would be great to see this integrated into the DWMBlurGlass GUI some day.

**Reply by TheInsane101** [reply]
Ok, I just tried it out, and there are some issues, like the extra decimals causing DWMBG to ignore the values completely, and the need to restart DWM before being able to apply the updated config. (Can DWMBG only use 2 or 3 decimal places?) I've reported them to Catman, who says he'll fix the decimal thing, but he says he still doesn't know how to apply the changes to the config without restarting DWM or needing to reinstall DWMBG manually. Do you know how?

**Reply by TorutheRedFox** [reply]
> The intensity slider on the CPL page has been broken since atleast some early version of Windows 10, it doesn't set any value. You can notice this by going into the registry and looking at ColorizationColor as you change the slider and save, it'll always be 6b, and once you re-enter the color settings page it resets its position.

it'd be cool to see this fixed
